sdks/python/apache_beam/runners/common.py

```diff
@@ -61,6 +61,9 @@
   from apache_beam.transforms import sideinputs
   from apache_beam.transforms.core import TimerSpec
 
+SplitResultType = Tuple[
```

Comment:
Feels like it's time to make a named tuple here...

Reply:
I agree it would help a bit.  I searched around for places where we access individual elements of this tuple, and there aren't many cases. Here's the one I found:

```python
  def bundle_application(self,
                         op,  # type: operations.DoOperation
                         primary  # type: common.SplitResultType
                        ):
    ((element_and_restriction, output_watermark),
     _) = primary
    return self.construct_bundle_application(
        op, output_watermark, element_and_restriction)
```

What name would you give to each item?

```python
SplitResult = NamedTuple(
    'SplitResult', [
        ('???', Tuple[WindowedValue, Optional[Timestamp]]), 
        ('???', Optional[Timestamp])
    ])
```

Reply:
Maybe WatermarkedElement and CallbackTimestamp? Talking to @boyuanzz this code is actively being developed, so it might make sense to hold off a bit. In particular, the splits are not entirely symmetric (e.g. a primary only ever has the WindowedValue, while the Residual is more properly a `Tuple[WindowedValue, Optiona[Timestamp], Optional[Timestamp]]`.